includes/internal/emails/class-email-repository.php

```diff
@@ -0,0 +1,178 @@
+ */
+class Email_Repository {
+
+	/**
+	 * Email identifier meta key.
+	 *
+	 * @var string
+	 */
+	private const META_IDENTIFIER  = '_sensei_email_identifier';
+
+	/**
+	 * Email type meta key.
+	 *
+	 * @var string
+	 */
+	private const META_TYPE        = '_sensei_email_type';
+
+	/**
+	 * Email description meta key.
+	 *
+	 * @param string
+	 */
+	private const META_DESCRIPTION = '_sensei_email_description';
+	
+	/**
+	 * Check if email exists for identifier.
+	 *
+	 * @param string $identifier Email identifier.
+	 *
+	 * @return bool
+	 */
+	public function has( string $identifier ): bool {
+		$query = new WP_Query(
+			[
+				'post_type' => Email_Post_Type::POST_TYPE,
+				'meta_key' => self::META_IDENTIFIER,
+				'meta_value' => $identifier,
+			]
+		);
+
+		return $query->post_count > 0;
+	}
+
+	/**
+	 * Create email for identifier.
+	 *
+	 * @param string $identifier  Email identifier.
+	 * @param array  $types       Email types.
+	 * @param string $subject     Email subject.
+	 * @param string $description Email description.
+	 * @param string $content     Email content.
+	 *
+	 * @return bool
+	 */
+	public function create( string $identifier, array $types, string $subject, string $description, string $content ): bool {
+		if ( $this->has( $identifier ) ) {
+			return false;
+		}
+
+		$email_data = [
+			'post_status'    => 'publish',
+			'post_type'      => Email_Post_Type::POST_TYPE,
+			'post_author'    => 1,
+			'post_title'     => $subject,
+			'post_content'   => $content,
+			'meta_input'     => [
+				self::META_IDENTIFIER  => $identifier,
+				self::META_DESCRIPTION => $description,
+			],
+		];
+
+		$email_id = wp_insert_post( $email_data );
+
+		foreach ( $types as $type ) {
+			add_post_meta( $email_id, self::META_TYPE, $type );
+		}
+
+		return $email_id > 0;
+	}
+
+	/**
+	 * Delete email for identifier.
+	 *
+	 * @param string $identifier Email identifier.
+	 *
+	 * @return bool
+	 */
+	public function delete( string $identifier ): bool {
+		$query = new WP_Query(
+			[
+				'select'     => 'ID',
+				'post_type'  => Email_Post_Type::POST_TYPE,
+				'meta_key'   => self::META_IDENTIFIER,
+				'meta_value' => $identifier,
+			]
+		);
+
+		if ( $query->post_count === 0 ) {
+			return false;
+		}
+
+		$email_id = $query->posts[0]->ID;
+
+		return (bool) wp_delete_post( $email_id, true );
+	}
+
+	/**
+	 * Get email for identifier.
+	 *
+	 * @param string $identifier Email identifier.
+	 *
+	 * @return \WP_Post|null
+	 */
+	public function get( string $identifier ) {
+		$query = new WP_Query(
+			[
+				'post_type'  => Email_Post_Type::POST_TYPE,
+				'meta_key'   => self::META_IDENTIFIER,
+				'meta_value' => $identifier,
+			]
+		);
+
+		if ( $query->post_count === 0 ) {
+			return null;
+		}
+
+		return $query->posts[0];
+	}
+
+	/**
+	 * Get all emails for type.
+	 *
+	 * @param string $type Email type.
+	 *
+	 * @return Email[]
+	 */
+	public function get_all( string $type, $limit = 10, $offset = 0 ): array {
```

Comment:
I suppose we use it inside our list table.

Reply:
Sounds good, but that query is different now.